wlib/LangRef.cpp

```diff
@@ -473,7 +473,26 @@ class CLangRef : public ILangRef, public CUnknown
 	{
 		if (m_typeStr.length() > 0 && boost::algorithm::equals(m_typeStr, _T("salt"))) 
 		{
-			return SCLEX_SALT;
+			if (boost::algorithm::equals(m_elementType, ATTRIBUTE_TYPE_SALT))
+			{
+				return SCLEX_SALT;
+			}
+			else if (boost::algorithm::equals(m_elementType, ATTRIBUTE_TYPE_ECM) || boost::algorithm::equals(m_elementType, ATTRIBUTE_TYPE_ESDL))
+			{
+				return SCLEX_ESDL;
+			}
+			else if (boost::algorithm::equals(m_elementType, ATTRIBUTE_TYPE_KEL))
+			{
+				return SCLEX_KEL;
+			}
+			else if (boost::algorithm::equals(m_elementType, ATTRIBUTE_TYPE_DUD))
+			{
+				return SCLEX_DUD;
+			}
+			else if (boost::algorithm::equals(m_elementType, ATTRIBUTE_TYPE_GENERAL))
+			{
+				return SCLEX_GENERAL;
+			}
```

Comment:
Not sure the other lang types should be in this PR - but its ok to leave them there.